Add Onion Service entry to Glossary (#128)
freedomofpress/securedrop-docs: docs/glossary.rst

------------------
The *Application Server* runs the SecureDrop application. This server hosts both
the website that sources access (the *Source Interface*) and the website that
journalists access (the *Journalist Interface*). Both are published

> through an *onion service* because Sources, journalists, and admins

Reviewer: Sources->sources (mid-sentence)

to this server, and they may only do so using Tor.

Onion Service
-------------

> Tor onion services provide anonymous inbound connections to websites and other servers exclusively over the Tor network. For example, SecureDrop uses onion services for the *Journalist Interface* and *Source Interface* websites, as well as for adminsitrative access to the servers in SSH-over-Tor mode.

Reviewer: adminsitrative-> administrative

Reviewer: It may be useful to cross-link Tor's own glossary entry here, which may also offer some inspiration for making the language more accessible:
https://support.torproject.org/onionservices/

Author: @eloquence I noticed that Tor's glossary doesn't apply the same [caution when linking onion addresses](https://github.com/freedomofpress/securedrop-docs/pull/128#discussion_r555407001).

> For example, the DuckDuckGo onion is https://... (link)

Because of that, I added the cross-reference in a separate commit that you can keep, squash or remove when performing the rebase.

> Onion services can be accessed by clicking a link or pasting the onion service address into Tor Browser. For example, `sdolvtfhatvsysc6l34d65ymdwxcujausv7k5jk4cy5ttzhjoi6fzvyd.onion <https://sdolvtfhatvsysc6l34d65ymdwxcujausv7k5jk4cy5ttzhjoi6fzvyd.onion>`__ is SecureDrop's onion service address.

Reviewer: We generally prefer not to link to onion services, because such links are clickable in a clearnet browser, which may cause DNS leakage before the request fails. While this is just an example address, we should follow the same convention here. Note that this example address is not reachable using `https://` (HTTPS is not required for onion services, which provide their own layer of encryption, though it can be useful as a way to assert ownership).

Reviewer: I would also change "SecureDrop's onion service address" to "the onion service address for the SecureDrop website", just to be minimally ambiguous (which SecureDrop?).

Author: :bulb: Oh, quite a few things here, thank you for the explanations @eloquence.

Onion Service v2
""""""""""""""""

> The third generation of onion services (v3) provides stronger cryptographic primitives (algorithms) than v2 onion services, and include redesigned protocols that guard against service information leaks on the Tor network.

Author: ```suggestion
The third generation of onion services (v3) provides stronger cryptographic primitives (algorithms) than v2 onion services, and includes redesigned protocols that guard against service information leaks on the Tor network.
```

💡 @ninavizz This can be a convenient way to amend copy. In the "Files changed" tab of this PR, near the beginning of any line, you can make a little "+" button appear to leave a comment. In the comment composer, there is an icon representing a file with a +/- symbol that allows to **insert a suggestion**. Clicking it copies the line as-is and allows you to edit it at will. Once you post the comment, a diff is automatically generated to highlight your changes 🙂 

It only works on line that have already been changed (that appear in the "Files changed" tab), but the result can be committed directly using the GitHub interface, so it's quite handy for larger re-wordings.

Author: **Extra level**: if you drag the little "+" icon when leaving a comment, you can leave a comment on multiple lines at once. ✨

Reviewer: (FYI, Nina is taking a break until later in January and may not be able respond here.)

IMO if we unpack "cryptographic primitives (algorithms)" we may as well just say "cryptographic algorithms".

Author: > Including words like "Cryptography Primitives" is discouraged, as so few users will understand what that means for the specificity to have value. I moved that bit to suggested text for the Onion Services glossary item. (@ninavizz's notes in #122)

I made the change to "cryptographic algorithms", though on second thought, I'm wondering if "**security features**" wouldn't be more accessible, even in the glossary.

> It is strongly recommended that only v3 onion services are used with SecureDrop 1.7.0 and forward.

Author: ```suggestion
It is strongly recommended that only v3 onion services are used from SecureDrop 1.7.0 onwards.
```

Reviewer: We can strengthen this recommendation now:

> Support for v2 onion services will start to be removed from SecureDrop in February 2021, and we strongly recommend [upgrading SecureDrop instances to v3 onion services](https://docs.securedrop.org/en/stable/v3_services.html) as soon as possible.
